### PR TITLE
SVG sprites: Proper bundling, Nunjucks "svg" macro

### PR DESCRIPTION
**Pull request by kettanaito**
The following changes were implemented:
* **Changed the logic behind globbing SVG sprites:** now we analyze nested folders under `${SVG_BASE}` and build the object in format `{ name: 'icons', glob: '/gfx/svg/icons/*.svg' }`. Then the same Gulp task runs through each collected object/sprite. This provides better control over globbing.
* Adjusted `svg()` macro in Nunjucks to be able to use it with default sprite (`icons` by default) or custom sprite for the projects were it is possible to have 2+ sprite bundles
* Added commentary to macro and `icons.nunj` pages, explaining the usage of the macro
* Uploaded new SVG icons for demonstration purpose

### CODE REVIEW

File: gulp/tasks/svg.js [hunk 1]
```diff
@@ -1,43 +1,63 @@
...
 const rename = require('gulp-rename');
-const browserSync = require('browser-sync');
 const svgmin = require('gulp-svgmin');
 const svgstore = require('gulp-svgstore');
 
 /* SVG sprites */
-/* Single SVG images are optimized in "images.js" */
+/* Returns an Object in a format { folderName: globbingPath } */
+function getSprites() {
+    var sprites = [];
```
Comment (janpanschab) [inline gulp/tasks/svg.js]: Should be possible use const here.

File: gulp/tasks/svg.js [hunk 2]
```diff
@@ -1,43 +1,63 @@
...
+
+    function getSpriteFolders(SVG_FOLDER) {
+        return fs.readdirSync(SVG_FOLDER).filter(file => fs.statSync(path.join(SVG_FOLDER, file)).isDirectory());
+    }
+    const spriteFolders = getSpriteFolders(config.SVG_BASE);
+
+    spriteFolders.map((spriteName) => {
```
Comment (janpanschab) [inline gulp/tasks/svg.js]: Use `.forEach` if you are not creating new array.

**Review by janpanschab — CHANGES_REQUESTED** [review]

File: gulp/tasks/svg.js [hunk 3]
```diff
@@ -57,7 +57,7 @@ gulp.task('svg:sprite', () => {
     }
 
     const sprites = getSprites();
-    return sprites.map(sprite => bundle(sprite));
+    return sprites.forEach(sprite => bundle(sprite));
```
Comment (janpanschab) [inline gulp/tasks/svg.js:60]: Should’t we merge the streams here? Or, is it OK to return `sprites`?
But if it works...
@vbulant

Comment (vbulant) [reply gulp/tasks/svg.js:60]: Good idea. I don’t have that much experience with streams, but I’d give it a try.
Another suggestion: when you use forEach or similar higher order functions, you can do just
```js
sprites.forEach(bundle);
```

**Review by janpanschab — APPROVED** [review]

**Comment by janpanschab** [comment]
I’ve already approved this PR? Do we want to merge it now?

**Comment by vbulant** [comment]
Well then I’d merge it.

**Pull request merged**